pkg/jobs/hive/hive.go

```diff
@@ -646,3 +994,113 @@ func retreiveAndUpdateClusterVersion(
 
 	return mcaStatus, nil
 }
+
+func eusRetreiveAndUpdateClusterVersion(
+	client clientv1.Client,
+	clusterName string,
+	updateVersion string,
+	managedclusterview *managedclusterviewv1beta1.ManagedClusterView,
+	isInterVersion bool) (managedclusteractionv1beta1.ManagedClusterAction, error) {
+
+	// Get latest clusterversion or else the object might be stale
+	mcview := managedclusterviewv1beta1.ManagedClusterView{}
+	mcaStatus := managedclusteractionv1beta1.ManagedClusterAction{}
+	resultmcview := managedclusterviewv1beta1.ManagedClusterView{}
+	clusterVersion := map[string]interface{}{}
```

Comment:
so I am guessing we can't use the proper type. just double checking.

Reply:
Right hive doesn't have the clusterversion. It's from Openshift.